test/trento/domain/cluster/cluster_test.exs

```diff
@@ -116,25 +116,165 @@ defmodule Trento.ClusterTest do
       )
     end
 
-    test "should return an error if the cluster was not registered yet and a command from a non-DC is received" do
-      assert_error(
-        [],
+    test "should add a host the cluster when the cluster has empty/unknown details" do
+      cluster_id = Faker.UUID.v4()
+      host_id = Faker.UUID.v4()
+      host_id_two = Faker.UUID.v4()
+
+      name = Faker.StarWars.character()
+
+      assert_events_and_state(
+        [
+          build(
+            :cluster_registered_event,
+            cluster_id: cluster_id,
+            name: name,
+            sid: nil,
+            provider: :unknown,
+            type: :unknown,
+            health: :unknown,
+            details: nil
+          ),
+          build(:host_added_to_cluster_event, cluster_id: cluster_id, host_id: host_id)
+        ],
         RegisterClusterHost.new!(%{
-          cluster_id: Faker.UUID.v4(),
-          host_id: Faker.UUID.v4(),
+          cluster_id: cluster_id,
+          host_id: host_id_two,
           name: Faker.StarWars.character(),
-          sid: Faker.StarWars.planet(),
-          discovered_health: :unknown,
+          sid: Faker.StarWars.character(),
           type: :hana_scale_up,
+          discovered_health: :unknown,
           designated_controller: false,
           provider: :azure
         }),
-        {:error, :cluster_not_found}
+        [
+          %HostAddedToCluster{
+            cluster_id: cluster_id,
+            host_id: host_id_two
+          }
+        ],
+        fn cluster ->
+          assert cluster_id == cluster.cluster_id
+          assert name == cluster.name
+          assert :unknown == cluster.type
+          assert :unknown == cluster.provider
+          assert 2 == cluster.hosts_number
+          assert [host_id_two, host_id] == cluster.hosts
+          assert :unknown == cluster.health
+        end
+      )
+    end
+
+    test "should create a cluster with empty details when the cluster was not registered yet and a command from a non-DC is received" do
```

Comment:
`should register a cluster...`
And it would make sense to put this test after the 1st one, as they are totally related. Like 1st we test the cluster registration from scratch, and them the adding of hosts

Reply:
done